src/jupyter_contrib_nbextensions/nbextensions/runtools/main.js

```diff
@@ -521,57 +630,63 @@ define([
-                    //cell.code_mirror.refresh();
+            if ((cell instanceof Jupyter.CodeCell)) {
+                var gutters = cell.code_mirror.getOption('gutters').slice();
+                if ($.inArray("CodeMirror-cellstate", gutters) < 0) {
+                    gutters.push('CodeMirror-cellstate')
+                    cell.code_mirror.setOption('gutters', gutters);
                 }
+                cell.code_mirror.on("change", changeEvent);
+                cell.code_mirror.refresh();
             }
         }
         /**
          * Restore hide/show status after reload
          */
         for (i=0; i<ncells; i++) {
-            var _cell=cells[i];
-            if (_cell.metadata.hasOwnProperty('hide_input') && _cell.metadata.hide_input === true )
-                showCell(_cell, 'i',false);
-            if (_cell.metadata.hasOwnProperty('hide_output') && _cell.metadata.hide_output === true )
-                showCell(_cell, 'o',false);
-            if (_cell.metadata.hasOwnProperty('locked') && _cell.metadata.locked === true ) {
-                _cell.code_mirror.setOption('readOnly', true);
-                _cell.code_mirror.setGutterMarker(0, "CodeMirror-foldgutter", makeLockMarker());
-                //_cell.code_mirror.refresh();
+            var cell=cells[i];
+            if (cell.metadata.hasOwnProperty('hide_input') && cell.metadata.hide_input === true )
+                showCell(cell, 'i',false);
+            if (cell.metadata.hasOwnProperty('hide_output') && cell.metadata.hide_output === true )
+                showCell(cell, 'o',false);
+            if (cell.is_editable() === false) {
+                set_cell_state(cell, 'locked');
             }
-        _cell.code_mirror.refresh();
+        cell.code_mirror.refresh();
         }
     };
 
     /**
-     * Called after extension was loaded
+     * Called from notebook after extension was loaded
      *
      */
     var load_extension = function() {
-        events.on('create.Cell',create_cell);
         load_css('./main.css');
-        load_css('codemirror/addon/fold/foldgutter.css');
-        load_css( './gutter.css'); /* change default gutter width */
-        if (Jupyter.notebook !== undefined && Jupyter.notebook._fully_loaded) {
-            require(['codemirror/addon/fold/foldgutter'], initGutter)
-        } else {
-            events.on("notebook_loaded.Notebook", function () {
-                require(['codemirror/addon/fold/foldgutter'], initGutter)
-            })
-        }
+        load_css('./gutter.css'); /* change default gutter width */
+        /* require our additional custom codefolding modes before initialising fully */
+        require(['./cellstate'], function () {
+            if (Jupyter.notebook._fully_loaded) {
+                initGutter();
+            }
+            else {
+                events.one('notebook_loaded.Notebook', initGutter);
+            }
+        });
+        config.load()
```

Comment:
this can be replaced with `Jupyter.notebook.config.loaded.then(initialize);` as discussed above